doc/specs/vulkan/chapters/memory.txt

```diff
@@ -552,57 +552,74 @@ There must: be at least one memory type with the
 ename:VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT bit set in its
 pname:propertyFlags.
 
-The memory types are sorted according to a preorder which serves to aid in
-easily selecting an appropriate memory type.
-Given two memory types X and Y, the preorder defines [eq]#X {leq} Y# if:
+For each pair of elements *X* and *Y* returned in pname:memoryTypes, *X*
+must: be placed at a lower index position than *Y* if:
+
+  * either the set of bit flags returned in the pname:propertyFlags member
+    of *X* is a strict subset of the set of bit flags returned in the
+    pname:propertyFlags member of *Y*.
+  * or the pname:propertyFlags members of *X* and *Y* are equal, and *X*
+    belongs to a memory heap with greater performance (as determined in an
+    implementation-specific manner).
 
-  * the memory property bits set for X are a strict subset of the memory
-    property bits set for Y. Or,
-  * the memory property bits set for X are the same as the memory property
-    bits set for Y, and X uses a memory heap with greater or equal
-    performance (as determined in an implementation-specific manner).
+[NOTE]
+.Note
+====
+There is no ordering requirement between *X* and *Y* that are not in a
+subset relation, which potentially allows more than one possible way to
```

Comment:
"...that are not in a subset relation" reads weirdly to me - how about "...where neither is a subset of the other"?

Reply:
Dunno, it is a mouthful.
It also does not make that much sense literally. **If** there was a requirement "where neither is a subset of the other", then there would be zero possible orders.

I.e.
If I added requirement:
- or the `propertyFlags` members of **X** and **Y** are neither subset of the other

I would not be able to sort (**a**,**b**) array, because **a** has to be before **b** and at the same time **b** has to be before **a**.

It probably needs to be reworded differently for the extra word-smithing points...